.github/workflows/ci.yml

```diff
@@ -234,15 +234,34 @@ jobs:
             cache: pip
         # setuptools is needed to get distutils on 3.12, which cythonize requires
       - name: install trio and setuptools
-        run: python -m pip install --upgrade pip . setuptools
+        run: python -m pip install --upgrade pip . setuptools 'coverage[toml]'
+
+      - name: add cython plugin to the coveragepy config
+        run: >-
+          sed -i 's#plugins\s=\s\[\]#plugins = ["Cython.Coverage"]#'
+          pyproject.toml
 
       - name: install cython & compile pyx file
+        env:
+          CFLAGS: ${{ env.CFLAGS }} -DCYTHON_TRACE_NOGIL=1
         run: |
           python -m pip install "cython${{ matrix.cython }}"
-          cythonize --inplace tests/cython/test_cython.pyx
+          cythonize --inplace -X linetrace=True tests/cython/test_cython.pyx
 
       - name: import & run module
-        run: python -c 'import tests.cython.test_cython'
+        run: coverage run -m tests.cython.run_test_cython
+
+      - name: get Python version for codecov flag
+        id: get-version
+        run: >-
+          echo "version=$(python -V | cut -d' ' -f2 | cut -d'.' -f1,2)"
+          >> "${GITHUB_OUTPUT}"
+      - if: always()
+        uses: codecov/codecov-action@v5
```

Comment:
is there any problem with mixing versions? I previously hit errors I thought was caused by that.

Reply:
No, they just use different uploader and APIs. But newer versions are less prone to flaky upload crashes for some reason, as I noticed.